### DISCUSSION TITLE
Looking for some security for hetzner_k3s

### DISCUSSION THREAD

**Discussion by Dani4kor** [vitobotta/hetzner-k3s#245] [General]
As subject says, is there some TIPS & TRICK or steps to add more security when creating or post creating cluster with hetnzer_k3s? as i see, we cant restrict IPS or add firewall to Hetzner LB with API or WEBUI which add our 6443 port to public. Do we have some steps to post creating to add layer of security with iptables or whatever? just thinking

Iam also looking for other small changes as creating nodes without public ip or only with ipv4 public in cluster creation step

Maybe someone have ideas, or solutions ?

Oh y, the tool is  awesome:) 

Thanks

**Comment by vitobotta** [comment]
Hi! The Hetzner firewall, which the tool sets up, is pretty solid. k3s has pretty good security defaults, so just make sure you restrict access to both SSH and the Kubernetes API by IPs and you are good to go. No need for any iptables configuration because the Hetzner firewall is applied to traffic even before it reaches your nodes. 

I have it on my list to add the ability to configure nodes without IP but no ETA at the moment.

> Oh y, the tool is awesome:)

Thanks! :)
